File: delegationadapter/src/main/java/com/kevin/delegationadapter/binding/BindingViewHolder.java
/*
 * Copyright (C) 2018 Baidu, Inc. All Rights Reserved.
 */

package com.kevin.delegationadapter.binding;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * BindingViewHolder
 *
 * @author zwenkai@foxmail.com, Created on 2018-04-03 17:51:49
 *         Major Function：<b>Binding ViewHolder</b>
 *         <p/>
 *         注:如果您修改了本类请填写以下内容作为记录，如非本人操作劳烦通知，谢谢！！！
 * @author mender，Modified Date Modify Content:
 */

public final class BindingViewHolder extends RecyclerView.ViewHolder {

    private ViewDataBinding binding;

    public BindingViewHolder(View itemView) {
        super(itemView);
    }

    public ViewDataBinding getBinding() {
        return binding;
    }

    public void setBinding(ViewDataBinding binding) {
        this.binding = binding;
    }
}

File: delegationadapter-extras/src/main/java/com/kevin/delegationadapter/extras/binding/BindingViewHolder.kt
/*
 * Copyright (c) 2018 Kevin zhou
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.kevin.delegationadapter.extras.binding

import android.view.View
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

/**
 * BindingViewHolder
 *
 * @author zwenkai@foxmail.com, Created on 2018-04-03 17:51:49
 *         Major Function：<b>Binding ViewHolder</b>
 *         <p/>
 *         Note: If you modify this class please fill in the following content as a record.
 * @author mender，Modified Date Modify Content:
 */
class BindingViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private var binding: ViewDataBinding? = null

    fun <T : ViewDataBinding> getBinding(): T {
        @Suppress("UNCHECKED_CAST")
        return binding as T
    }

    fun setBinding(binding: ViewDataBinding) {
        this.binding = binding
    }
}
